src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs

```diff
@@ -1992,6 +2054,117 @@ public void ChainTHasCheckpointAt1000_MaxReorgIs10_TwoChainsPriorTo1000Presented
             consumedHeaders = connectionResult.Consumed.ToArray(listOfChainAHeaders.Count);
             consumedHeaders.HaveBlockDataAvailabilityStateOf(BlockDataAvailabilityState.HeaderOnly).Should().BeTrue();
         }
+      
+        /// <summary>
+        /// Issue 19 @ New peer K claims 10d. Peer K disconnects. Chain shouldn't change.
+        /// </summary>
+        [Fact]
+        public void NewPeerClaimsHeadSecondTime_PeerDisconnected_ChainShouldNotChange()
+        {
+            const int initialChainSize = 5;
+            TestContext ctx = new TestContextBuilder().WithInitialChain(initialChainSize).UseCheckpoints().Build();
+            ChainedHeaderTree cht = ctx.ChainedHeaderTree;
+            ChainedHeader initialChainTip = ctx.InitialChainTip;
+
+            IEnumerable<BlockHeader> listOfAllUniqueBlockHeaders = ctx.SetupPeersForTest(initialChainSize, cht, initialChainTip, out ChainedHeader chainATip, out ChainedHeader chainDTip);
+
+            // Additional SetUp for current test.
+            ChainedHeader chainKTip = chainDTip; //peer K has exactly the same chain as peer D.
+            List<BlockHeader> peerKBlockHeaders = ctx.ChainedHeaderToList(chainKTip, chainKTip.Height);
+
+            // Claiming Peer K chain.
+            ConnectNewHeadersResult connectionPeerKResult = cht.ConnectNewHeaders(4, peerKBlockHeaders);
+            ctx.SwitchToChain(cht, chainKTip, connectionPeerKResult.Consumed, 5);
+
+            var chainedHeadersWithPeerK = new Dictionary<uint256, ChainedHeader>(cht.GetChainedHeadersByHash());
+
+            cht.PeerDisconnected(4);
+
+            Dictionary<uint256, ChainedHeader> chainedHeadersWithoutPeerK = cht.GetChainedHeadersByHash();
+
+            chainedHeadersWithoutPeerK.Should().BeEquivalentTo(chainedHeadersWithPeerK);
+
+            this.CheckChainedHeaderTreeConsistency(cht, listOfAllUniqueBlockHeaders);
+        }
+
+        /// <summary>
+        /// Basic test for tests 18 to 20, 28.
+        /// 1/6 There is no branch which is claimed by no one (PeerTipsByHash have tip of every branch), none of list should be empty.
+        /// 2/6 ChainHeadersByHash contains only reachable headers.
+        /// 3/6 ChainHeader claims LocalPeer.
+        /// 4/6 Each connected peer has exactly 1 entry in PeertipsByHash.
+        /// 5/6 CH.Next[i].Prev == CH for every header and every .Next.
+        /// 6/6 PeerTipsByPeerId should reflect PeerIdsByHash(except local marker).
+        /// </summary>
+        /// <param name="cht">ChainHeaderTree.</param>
+        /// <param name="listOfAllUniqueBlockHeaders">List of All unique blockheader in all connected chains.</param>
+        private void CheckChainedHeaderTreeConsistency(ChainedHeaderTree cht, IEnumerable<BlockHeader> listOfAllUniqueBlockHeaders)
+        {
+            bool noUnclaimedBraches = true;
+            bool eachPeerOneEntry = true;
+            bool reflecttipHash = true;
+            bool chbhHasOnlyReachebleHeaders = true;
```

Comment:
Spelling, but also you don't need these Boolean variables here or maybe at all if you just assert immediately when you send them to false